libs/langchain/langchain/embeddings/azure_openai.py

```diff
@@ -0,0 +1,149 @@
+"""Azure OpenAI embeddings wrapper."""
+from __future__ import annotations
+
+import os
+import warnings
+from typing import Dict, Optional, Union
+
+from langchain.embeddings.openai import OpenAIEmbeddings
+from langchain.pydantic_v1 import Field, root_validator
+from langchain.utils import get_from_dict_or_env
+from langchain.utils.openai import is_openai_v1
+
+
+class AzureOpenAIEmbeddings(OpenAIEmbeddings):
+    """`Azure OpenAI` Embeddings API."""
+
+    azure_endpoint: Union[str, None] = None
+    """Your Azure endpoint, including the resource.
+
+        Example: `https://example-resource.azure.openai.com/`
+    """
+    azure_deployment: Optional[str] = None
+    """A model deployment. 
+
+        If given sets the base client URL to include `/deployments/{azure_deployment}`.
+        Note: this means you won't be able to use non-deployment endpoints.
+    """
+    openai_api_key: Union[str, None] = Field(default=None, alias="api_key")
+    """Automatically inferred from env var `AZURE_OPENAI_API_KEY` if not provided."""
+    azure_ad_token: Union[str, None] = None
+    """Your Azure Active Directory token.
+
+        Automatically inferred from env var `AZURE_OPENAI_AD_TOKEN` if not provided.
+
+        For more: 
+        https://www.microsoft.com/en-us/security/business/identity-access/microsoft-entra-id.
+    """  # noqa: E501
+    azure_ad_token_provider: Union[str, None] = None
+    """A function that returns an Azure Active Directory token.
+
+        Will be invoked on every request.
+    """
+    openai_api_version: Optional[str] = Field(default=None, alias="api_version")
+    """Automatically inferred from env var `OPENAI_API_VERSION` if not provided."""
+    validate_base_url: bool = True
```

Comment:
Could we steal doc-string for this? (No idea what it does otherwise)

Reply:
this is ours not from openai, used for backwards compatibility